lib/adapters/autocomplete-adapter.ts

```diff
@@ -429,18 +451,29 @@ export default class AutocompleteAdapter {
    * @param textEdit A {TextEdit} from a CompletionItem to apply.
    * @param editor An Atom {TextEditor} used to obtain the necessary text replacement.
    * @param suggestion An {atom$AutocompleteSuggestion} to set the replacementPrefix and text properties of.
+   * @param insertMode The behavior of suggestion acceptance (see {InsertMode}).
    */
   public static applyTextEditToSuggestion(
-    textEdit: TextEdit | undefined,
+    textEdit: TextEdit | InsertReplaceEdit | undefined,
     editor: TextEditor,
     triggerColumns: [number, number],
     originalBufferPosition: Point,
     suggestion: TextSuggestion,
+    insertMode: InsertMode,
   ): void {
     if (!textEdit) { return; }
-    if (textEdit.range.start.character !== triggerColumns[0]) {
-      const range = Convert.lsRangeToAtomRange(textEdit.range);
-      suggestion.customReplacmentPrefix = editor.getTextInBufferRange([range.start, originalBufferPosition]);
+    let range: Range;
+    if ('range' in textEdit) {
+      range = textEdit.range;
+    } else if (insertMode === InsertMode.insert) {
+      range = textEdit.insert;
+    } else {
+      range = textEdit.replace;
+    }
+
+    if (range.start.character !== triggerColumns[0]) {
+      const atomRange = Convert.lsRangeToAtomRange(range);
+      suggestion.customReplacmentPrefix = editor.getTextInBufferRange([atomRange.start, originalBufferPosition]);
```

Comment:
This code needs a test

Reply:
I am, but seeing as I'm a little busy right now, it might take a couple days before I can get to it. If that's fine with you, then sure, I'll add it.